Scripts/Modules/SpatialHintModule/SpatialHintModule.cs

```diff
@@ -32,65 +31,65 @@ public SpatialHintStateFlags state
 				switch (m_State)
 				{
 					case SpatialHintStateFlags.Hidden:
-						m_SpatialHintModuleUI.centeredScrolling = false;
-						m_SpatialHintModuleUI.preScrollArrowsVisible = false;
-						m_SpatialHintModuleUI.secondaryArrowsVisible = false;
-						this.RemoveRayVisibilitySettings(m_ControllingRayOrigin, this);
-						controllingRayOrigin = null;
+						m_SpatialHintUI.centeredScrolling = false;
+						m_SpatialHintUI.preScrollArrowsVisible = false;
+						m_SpatialHintUI.secondaryArrowsVisible = false;
+						this.RemoveRayVisibilitySettings(this.RequestRayOriginFromNode(m_ControllingNode), this);
```

Comment:
Should we cache the ray origin when a controlling node is set?  Could a situation arise where this result changes between Add and Remove VisibilitySettings, and in that situation would we want different ray origins set?